pkg/config/client/client_config.go

```diff
@@ -21,8 +21,11 @@ func GetKubeConfigOrInClusterConfig(kubeConfigFile string, overrides *ClientConn
 	if err != nil {
 		return nil, err
 	}
+
 	applyClientConnectionOverrides(overrides, clientConfig)
-	clientConfig.WrapTransport = defaultClientTransport
+
+	t := clientTransportOverrides{maxIdleConnsPerHost: overrides.MaxIdleConnsPerHost}
```

Comment:
why this line? why not assign directly?

Reply:
ic, defaultClientTransport is a method. I prefer the functional style `func defaultClientTransport(maxIdleConnsPerHost) func (...) {...}`. But @deads2k hates it. Nvmd.

Reply:
funny, i had using functional style and I hated it because it was pretty hard to read ;-)

Reply:
I am used to scheme, haskell, scala and friends. Every method looks hard to read ;-)